src/utils/hub-channel.js

```diff
@@ -145,14 +145,12 @@ export default class HubChannel extends EventTarget {
     // Note: token is not verified.
     this._permissions = jwtDecode(token);
     configs.setIsAdmin(this._permissions.postgrest_role === "ret_admin");
-    this.dispatchEvent(new CustomEvent("permissions_updated"));
+    this.dispatchEvent(new CustomEvent("permissions_updated", { detail: { permsToken: token } }));
 
     // Refresh the token 1 minute before it expires.
-    const nextRefresh = new Date(this._permissions.exp * 1000 - 60 * 1000) - new Date();
-    setTimeout(async () => {
-      const result = await this.fetchPermissions();
-      this.dispatchEvent(new CustomEvent("permissions-refreshed", { detail: result }));
```

Comment:
`fetchPermissions` calls `setPermissionsFromToken` which dispatches `permissions_updated` so we don't need to dispatch here.